### PR TITLE
unity tests: return non zero exit code when test failed

### PR DESCRIPTION
**Pull request by mateusz-bloch**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: devices/grlib-multi/test-grlib-multi.c [hunk 1]
```diff
@@ -775,6 +775,5 @@ void runner(void)
 
 int main(int argc, char *argv[])
 {
-	UnityMain(argc, (const char **)argv, runner);
-	return 0;
+	return UnityMain(argc, (const char **)argv, runner) == 0 ? EXIT_SUCCESS : EXIT_FAILURE;
```
Comment (damianloew) [inline devices/grlib-multi/test-grlib-multi.c:778]: The problem is still not resolved, please check the following code and just run in on host:

```C
#include <stdio.h>
#include <stdlib.h>

int main(void)
{
	return 256 == 0 ? EXIT_SUCCESS : EXIT_FAILURE;
}
```

Comment (mateusz-bloch) [reply devices/grlib-multi/test-grlib-multi.c:778]: In the scenario you provided, it will always return 1, because of EXIT_FAILURE

Comment (damianloew) [reply devices/grlib-multi/test-grlib-multi.c:778]: oh, that's right - it seems to work fine

File: devices/grlib-multi/test-grlib-multi.c [hunk 2]
```diff
@@ -775,6 +775,5 @@ void runner(void)
```
Comment (damianloew) [inline devices/grlib-multi/test-grlib-multi.c:775]: Please correct it also in waitpid test. We should also consider corrections in other repos - for example phoenix-rtos devices

File: devices/grlib-multi/test-grlib-multi.c (hunk 1 above)
Comment (damianloew) [inline devices/grlib-multi/test-grlib-multi.c:778]: I know that this version is also correct, but to make it more clear I would add parentheses:
```suggestion
	return (UnityMain(argc, (const char **)argv, runner) == 0) ? EXIT_SUCCESS : EXIT_FAILURE;
```

**Review by damianloew — APPROVED** [review]
LGTM

**Pull request merged**